source/scpp/extra/DUtils.h

```diff
@@ -27,6 +27,23 @@ int cpp_set_foreach(void* setptr, void* ctx, void* func)
     return 0;
 }
 
+// walking through an std::set with index support
+template<typename T>
+int cpp_set_foreach_ind(void* setptr, void* dg)
```

Comment:
Why `void*` ? You should be able to pass the real types, wouldn't you ?

Reply:
this method is really just a copy of cpp_set_foreach without the wrapper. Do you remember why void was used there? 

![image](https://user-images.githubusercontent.com/19815616/116834189-ead48480-abf7-11eb-925e-6b44ae2f3d8e.png)

Reply:
It was due to linker errors when the right type was used.

Reply:
seems they fixed the compiler, it works now

Reply:
now it crashed dmd, will nuke this commit until the dmd is fixed
https://issues.dlang.org/show_bug.cgi?id=21887#c0